src/main/scala/rugds/sbt/DockerDevEnvironment.scala

```diff
@@ -0,0 +1,177 @@
+        }
+        .map {
+          // replace project dependencies (with no jar) with its jar equivalents
+          case (module, dependencies) =>
+            val m = projectJarDependencies.find(_ == module).getOrElse(module)
+            val d = dependencies.map(dm => projectJarDependencies.find(_ == dm).getOrElse(dm))
+            m -> d
+        }
+        .filter {
+          // remove projects without jar
+          case (module, dependencies) => module.jarFile.isDefined
+        }
+        .map {
+          // remove projects without jar in dependencies
+          case (module, dependencies) => module -> dependencies.filter(_.jarFile.isDefined)
+        }
+      jarDependencies.map {
+        case (module, dependencies) => if (module == projectId) module -> ((jarDependencies.toMap.keySet ++ dependencies) - module) else module -> dependencies
+      }.sortWith {
+        case ((m1, d1), (m2, d2)) => d2.contains(m1) || !d1.contains(m2)
+      }
+    },
+
+    envDockerDependencyTree := envProjectDependencyTree.value.flatMap {
+      case (module, dependencies) => module.toEnvDocker.map(_ -> dependencies.flatMap(_.toEnvDocker))
+    },
+
+    envPullDockerImages := {
+      val log = streams.value.log
+      envDockerDependencyTree.value.map {
+        case (image, _) => image.toString
+      }
+      .map { image => {
+        log.info(s"docker images -q $image")
+        if (s"docker images -q $image".!!.isEmpty) {
+          log.info(s"docker pull $image")
+          if (s"docker pull $image".! != 0) {
+            throw new IllegalStateException(s"Cannot found or pull docker image: $image")
+          }
+        }
+        image
+      }}.toSet
+    },
+
+    envStartDev := {
+      val log = streams.value.log
+      val pdi = envPullDockerImages.value
+//      val networkCmd = s"docker -P" // TODO: auto-create network if not yet created (or fail if exists?)
+      val networkId = "rugds-dev"
+      envDockerDependencyTree.value.map {
+        case (docker, dependencies) =>
+          val dockerPsCmd = Seq("docker", "ps", "-aq", "--format", "{{ .Names}}"); log.info(dockerPsCmd.mkString(" "))
+          if (!dockerPsCmd.lines.contains(docker.name)) { // if it does not exist => run it
+            val dockerInspectCmd = Seq("docker", "inspect", "-f", "{{range $key, $item := .Config.ExposedPorts}}{{$key}} {{end}}", docker.toString); log.info(dockerInspectCmd.mkString(" "))
+            val dockerExposedPorts = dockerInspectCmd.!!
+            val exposedPorts = dockerExposedPorts.trim.split(' ').map(port => port.split('/')).map(list => (list.head, list.last)).map {
+              case (port, _) => s"-p $port:$port"
+            }.mkString(" ")
+            val dockerRunCmd = s"docker run -d --network=$networkId $exposedPorts --name ${docker.name} ${docker.toString}"; log.info(dockerRunCmd)
+            dockerRunCmd.!!
+          }
+          docker
+      }.toSet
+    },
+
+    envDockerList := (resources in Compile).value
```

Comment:
Is this only for the docker dependencies inside the current project, or also for external?